projects/js-packages/publicize-components/src/social-store/actions/scheduled-shares.ts

```diff
@@ -1,6 +1,25 @@
 import { store as coreStore } from '@wordpress/core-data';
 import { __ } from '@wordpress/i18n';
 import { store as noticesStore } from '@wordpress/notices';
+import { ScheduledShare } from '../types';
+
+/**
+ * Creates a new scheduled share.
+ *
+ * @param data - The data.
+ *
+ * @return A thunk.
+ */
+export function createScheduledShare(
+	data: Pick< ScheduledShare, 'post_id' | 'connection_id' | 'timestamp' > &
+		Partial< Pick< ScheduledShare, 'message' > >
+) {
+	return async function ( { registry } ) {
+		const { saveEntityRecord } = registry.dispatch( coreStore );
+
+		await saveEntityRecord( 'wpcom/v2', 'publicize/scheduled-actions', data );
```

Comment:
You can do it this way:

```suggestion
		const { saveEntityRecord } = registry.dispatch( coreStore );
		const { getLastEntitySaveError } = registry.select( coreStore );
		const { createErrorNotice, createSuccessNotice } = registry.dispatch( noticesStore );

		const success = await saveEntityRecord( 'wpcom/v2', 'publicize/scheduled-actions', data );

		// If the creation was not successful, show an error notice.
		if ( ! success ) {
			const lastError = getLastEntitySaveError( 'wpcom/v2', 'publicize/scheduled-actions' );

			let message = __( 'There was an error scheduling the post.', 'jetpack-publicize-components' );

			if ( lastError?.message ) {
				message += ' ' + lastError.message;
			}

			createErrorNotice( message, {
				type: 'snackbar',
			} );
		} else {
			createSuccessNotice( __( 'Post scheduled successfully.', 'jetpack-publicize-components' ), {
				type: 'snackbar',
			} );
		}
```

Reply:
@manzoorwanijk  I did this, but then reverted the change because this way we create a notice for each schedule, and if they schedule for 5 connections it results in 5 notices

Reply:
That's easy to fix. Just pass an `id` to both of those success and failure action calls

```js
{
    type: 'snackbar',
    id: 'social-scheduled-share',
}
```

That way, there should be just one notice.

Reply:
be1e5f6